src/code/code_800F9280.c

```diff
@@ -1,350 +1,388 @@
-                // with channel mask ~channelMask...
-                Audio_QueueCmdU16(0x90000000 | _SHIFTL(playerIdx, 16, 8), channelMaskReversed);
-                // unstop channels
-                Audio_QueueCmdS8(0x08000000 | _SHIFTL(playerIdx, 16, 8) | 0xFF00, 0);
+        case SEQCMD_OP_SET_CHANNEL_DISABLE_MASK:
+            // Disable or Reenable channels
+
+            // Disable channels
+            channelMaskDisable = cmd & 0xFFFF;
+            if (channelMaskDisable != 0) {
+                // Apply channel mask `channelMaskDisable`
+                Audio_QueueCmdU16(0x90000000 | _SHIFTL(seqPlayerIndex, 16, 8), channelMaskDisable);
+                // Disable channels
+                Audio_QueueCmdS8(0x08000000 | _SHIFTL(seqPlayerIndex, 16, 8) | 0xFF00, 1);
+            }
+
+            // Reenable channels
+            channelMaskEnable = (channelMaskDisable ^ 0xFFFF);
+            if (channelMaskEnable != 0) {
+                // Apply channel mask `channelMaskEnable`
+                Audio_QueueCmdU16(0x90000000 | _SHIFTL(seqPlayerIndex, 16, 8), channelMaskEnable);
+                // Enable channels
+                Audio_QueueCmdS8(0x08000000 | _SHIFTL(seqPlayerIndex, 16, 8) | 0xFF00, 0);
             }
             break;
 
-        case 0xB:
-            // update tempo
-            D_8016E750[playerIdx].unk_14 = cmd;
+        case SEQCMD_OP_TEMPO_CMD:
+            // Update a tempo using a sub-command system.
+            // Stores the cmd for processing elsewhere.
+            gActiveSeqs[seqPlayerIndex].tempoCmd = cmd;
             break;
 
-        case 0xC:
-            // start sequence with setup commands
+        case SEQCMD_OP_SETUP_CMD:
+            // Queue a sub-command to execute once the sequence is finished playing
             subOp = (cmd & 0xF00000) >> 20;
-            if (subOp != 0xF) {
-                if (D_8016E750[playerIdx].unk_4D < 7) {
-                    found = D_8016E750[playerIdx].unk_4D++;
+            if (subOp != SEQCMD_SUB_OP_SETUP_RESET_SETUP_CMDS) {
+                // Ensure the maximum number of setup commands is not exceeded
+                if (gActiveSeqs[seqPlayerIndex].setupCmdNum < 7) {
+                    found = gActiveSeqs[seqPlayerIndex].setupCmdNum++;
                     if (found < 8) {
-                        D_8016E750[playerIdx].unk_2C[found] = cmd;
-                        D_8016E750[playerIdx].unk_4C = 2;
+                        gActiveSeqs[seqPlayerIndex].setupCmd[found] = cmd;
+                        // Adds a delay of 2 frames before executing any setup commands.
+                        // This allows setup commands to be requested along with a new sequence on a seqPlayerIndex.
+                        // This 2 frame delay ensures the player is enabled before attempting to execute commands.
+                        // Otherwise, the setup commands will execute before the sequence starts instead of when
+                        // the sequence ends.
```

Comment:
Yeah, the intent of this is to be used with fanfares, which only play for a short amount of time.

Once the fanfare ends, you usually want things to happen (which sequence do you restore)?

But by setting up the "setup" commands at the beginning when you call the fanfare, you don't need to keep track of when the fanfare ends. Instead of checking if the fanfare is finished every frame manually, you just queue what the behavior is at the start so you take care of it all in a single function. Does that logic make sense?